### PR TITLE
Denis Charter Rewrite

### PR DESCRIPTION
**Pull request by OR13**
- https://mailarchive.ietf.org/arch/msg/spice/PAzTWaHdhIs-2-YtS62yi5nEoCI/

### CODE REVIEW

File: charter.md
Comment (OR13) [inline charter.md:7]: Digital credentials are essential to identity, authorization, licenses, certificates and digitization use cases that are part of modernization efforts targeting efficiency, transparency, such as digital licenses or certificates of origin.

Comment (OR13) [inline charter.md]:
```suggestion
Digital credentials are essential to identity, authorization, licenses, certificates and digitization use cases that are part of modernization efforts targeting efficiency, transparency, such as digital licenses or certificates of origin.
```

Comment (OR13) [reply charter.md]: I don't support framing the work as "for natural persons only", I would like to revert this set of changes to what we had previously.

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
In some contexts, holders may be willing either to partially disclose some values of their attributes or to demonstrate some properties about their attributes without disclosing their values.
```

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
When received by an entity, a proof of the digital credential needs to be provided and verified, so that only the legitimate controller of the digital credential can take an advantage of its possession. The use of Trusted Applications (TAs) supported by a Trusted Execution Environment (TEE) is essential in order to provide a proof of control of a digital credential.
```

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
Some holders may wish to carry more than one credential. These credentials, together with associated key material, can be stored in a identity digital wallet.
```

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
```

Comment (OR13) [reply charter.md]: This is not an accurate statement, and I am not sure we need to repeat what other working groups outside of IETF are doing, beyond what you have said above.

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
- Unlinkability (preventing tracking while proving attributes / capabilities)
- Selective Disclosure (supporting user agency, data minimization, redaction)
- Authenticity (ensuring information is trustworthy and can be authenticated)
- Confidentiality (ensuring information is protected from unauthorized access)
- Hardware Assurance (awareness of the security capabilities of issuers, holders and verifiers)
- Availability (efficiently information processing, minimizing data in flight, reducing carbon cost for storage and transmission, and supporting offline / paper credential use cases)
- Semantic interchangeability (ensuring terms are understood and used consistently, in a global or local context)
- Usability and Feasibility (the core operations must be easy to understand and use, and possible to integrate with existing platforms)
- Accountability (what format and protocol considerations can help mitigate abusive or excessive requests for credential presentation?  What mitigates abuse by issuers?)
```

Comment (OR13) [reply charter.md]: I prefer the framing we have had for this section previously.

File: charter.md
Comment (OR13) [inline charter.md:40]: good addition

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
Feedback from experts in other IETF working groups is gathered in the SPICE WG without creating fragments of credential work spread across several existing places in the IETF. Additionally, the SPICE WG will coordinate with other SDOs, such as ISO or W3C, on data model elements or protocols needed to support existing credential use cases.
```

Comment (OR13) [reply charter.md]: just fixing formatting

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
The SPICE WG will develop a framework in support of the 'Three Role Model', considering in particular selective disclosure of attributes and unlinkability between verifiers. It will consider the use of TEEs (Trusted Execution Environments) for managing key material and digital credentials.
```

Comment (OR13) [reply charter.md]: Just formatting

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
```

Comment (OR13) [reply charter.md]: This change is critical for my approval of this charter rewrite.

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
## Milestones
```

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
- A meta-data discovery document enabling the 3 roles (issuers, holders and verifiers) to discover supported protocols and formats for keys, claims, credentials and proofs.
```

Comment (OR13) [reply charter.md]: You can't omit issuers here, or its impossible for a verifier to check a proof.

File: charter.md
Comment (OR13) [reply charter.md]: formatting

File: charter.md
Comment (OR13) [inline charter.md:74]: I am fine with this.

File: charter.md
Comment (aj-stein-nist) [reply charter.md:10]: Is "use of Trusted Applications (TAs) supported by a Trusted Execution Environment (TEE) is essential in order to provide a proof of control of a digital credential" essential at this juncture and explicit in the charter? I know of one or two parties are assertive on this. I do not think this is a hard requirement for the charter. Should proponents work in this WG, TEEP, and other relevant groups for *later* documents? Sure. Does this mean all architectures and related deliverables must mandate TEE out the gate? It is not 100% so in this section, but we are getting close enough. Is that the most important thing now?

File: charter.md
Comment (aj-stein-nist) [reply charter.md]: I concur that natural persons frames this narrowly in a way I do not think is productive for the charter and this group at this juncture. Given this perspective, I only prefer the proposed change by Orie in the counter-edit in green.

File: charter.md
Comment (aj-stein-nist) [reply charter.md]: Agree with this change.

File: charter.md
Comment (aj-stein-nist) [reply charter.md]: Is "use of Trusted Applications (TAs) supported by a Trusted Execution Environment (TEE) is essential in order to provide a proof of control of a digital credential" essential at this juncture to be explicit in the charter? I know of one or two parties are assertive on this in the mailing list. I do not think this is a hard requirement for the charter. Should proponents work in this potential WG, TEEP, and other relevant groups for *later* documents? Sure. Does this mean all architectures and related deliverables must mandate TEE out the gate? It is not 100% so in this section, but we are getting close enough I am not sure. Is that the most important thing now?

I would disagree to adding this, but I want to hear feedback from others to know if I am the squeaky wheel minority here.

File: charter.md
Comment (aj-stein-nist) [reply charter.md]: I agree. Once we are talking about JSON-LD properties in the charter it is odd to me why we get this deep in the weeds (I am newer to IETF, but the other charters I have reviewed thus far do not).

File: charter.md
Comment (aj-stein-nist) [reply charter.md]: I agree with this change.

**Review by aj-stein-nist — COMMENTED** [review]
I have one open question about the implied importance of TEE, other than that I like all the changes, or I'd approve outright.

**Review by mprorock — CHANGES_REQUESTED** [review]
I don't see this as an improvement.

File: charter.md
Comment (mprorock) [reply charter.md]: I don't think we need this language - feels overcomplicated for the charter and too specific

File: charter.md
Comment (mprorock) [reply charter.md]: I don't think getting into this level of detail is helpful

File: charter.md
Comment (mprorock) [reply charter.md]: Agreed - let's keep the original framing

**Comment by mprorock** [comment]
I am happy to approve if we accept @OR13 suggestions save where noted that original sections were better

File: charter.md
Comment (Denisthemalice) [reply charter.md]: > I don't support framing the work as "for natural persons only", I would like to revert this set of changes to what we had previously.

However, no rational for this rejection has been provided. So I can only wonder why. 
Is there an intent to address legal persons as well ? 
If it is the case, I wonder why legal persons would have privacy concerns. 
Natural persons have privacy concerns, why would it also apply to legal persons ?
If it is not the case of legal persons, why other case(s) is it ?

File: charter.md
Comment (Denisthemalice) [reply charter.md]: The use of a TA (under a TEE) is necessary in order to prevent a collusion / collaborative attack between Alice and Bob.

As an example if Alice (12) is colluding with Bob (25), whatever cryptography is being used, it is impossible to prevent Alice to get some data from Bob demonstrating that she is over 18.

Preventing the export of the value of a private key or a secret key, does not prevent its use, at least, by the legitimate owner of these keys. At the time a credential request is initiated, the holder must demonstrate either that a private key or a link secret is stored in a TEE and originates from a given TA. 

Then after, these keys can only be used by that TA that does not allow any operation to be done with these keys, but only specific operations on already formatted data.

Without TEEs, collusion / collaborative attacks cannot be prevented.

File: charter.md
Comment (Denisthemalice) [reply charter.md]: The additions make clear the topics that will need to be addressed

File: charter.md
Comment (Denisthemalice) [reply charter.md]: Removing that sentence completely would be an easy way to solve the concern.

File: charter.md
Comment (mkhraisha) [reply charter.md]: Digital Credentials may be used to represent assets or organizations, not just natural persons, there should be no need to restrict it to just natural persons

Comment (brentzundel) [reply charter.md]: Mesur is using digital credentials for many use cases where they do not represent natural persons. Restricting things in this manner is not necessary.

File: charter.md
Comment (brentzundel) [reply charter.md]: I might be okay with "may be necessary", but I don't think this is a needed restriction.

Comment (brentzundel) [inline charter.md]:
```suggestion
When received by an entity, a proof of ownership of the digital credential needs to be provided and verified, so that only the legitimate owner of the digital credential can take an advantage of its possession. The use of Trusted Applications (TAs) supported by a Trusted Execution Environment (TEE) may be essential for some use cases in order to provide a proof of ownership of a digital credential.
```

File: charter.md
Comment (brentzundel) [inline charter.md]:
```suggestion
When received by an entity, a proof of the digital credential needs to be provided and verified, so that only the legitimate controller of the digital credential can take an advantage of its possession. The use of Trusted Applications (TAs) supported by a Trusted Execution Environment (TEE) may be essential for some use cases in order to provide a proof of control of a digital credential
```

File: charter.md
Comment (OR13) [inline charter.md]:
```suggestion
```

**Comment by OR13** [comment]
The call ended at this point, as I mentioned on the call, I will use my editor power to resolve the remaining comments, prepare a new draft charter, and submit a bof request.

**Comment by OR13** [comment]
All suggestions have been applied, i will do a once over to review before merging.

File: charter.md
Comment (brentzundel) [reply charter.md]: I think I understand what Denis is getting at, which is that Verifiers and Holder will need to discover metadata, but he is asserting that an Issuer will not. He may be correct, but it feels like an unnecessary limitation.
E.g., what if the issuer is a website and the browser is serving as the Holder wallet? It may be the most efficient protocol for the Issuer to discover metadata about the Holder's capabilities rather than the other way around.

**Pull request merged**
